tools/as/as.cpp

```diff
@@ -108,6 +103,16 @@ int main(int, const char** argv) {
   std::vector<char> contents;
   if (!ReadTextFile<char>(inFile.c_str(), &contents)) return 1;
 
+  if (target_env == spv_target_env::SPV_ENV_MAX) {
```

Comment:
If I understand this correctly, if the target is not specified on the comand line, and not inferred from comments, then this will cause an error.

This is a regression.  It should do what it did before: assume the latest SPIR-V version, and no target OS.